tests/packages/test_locker.py

```diff
@@ -622,6 +622,38 @@ def test_reading_lock_file_should_raise_an_error_on_invalid_data(
     assert "Unable to read the lock file" in str(e.value)
 
 
+def test_reading_lock_file_should_raise_an_error_on_missing_metadata(
+    locker: Locker,
+) -> None:
+    content = f"""\
+# {GENERATED_COMMENT}
+
+[[package]]
+name = "A"
+version = "1.0.0"
+description = ""
+optional = false
+python-versions = "*"
+files = []
+
+[package.source]
+type = "legacy"
+url = "https://foo.bar"
+reference = "legacy"
+"""
+    with locker.lock.open("w", encoding="utf-8") as f:
+        f.write(content)
+
+    with pytest.raises(RuntimeError) as e:
+        _ = locker.lock_data
+
+    assert (
```

Comment:
I'd run the assert in the `with pytest.raises` block.

Reply:
good catch, fixed!

Reply:
IMO, only the code that is expected to raise the exception should be in the `with pytest.raises`. That's also how we did it in all other tests, so please revert.

Reply:
reverted!